spec/unit/module_spec.rb

```diff
@@ -3,20 +3,36 @@
 
 describe R10K::Module do
   describe 'delegating to R10K::Module::Git' do
-    it "accepts args {:git => 'git url}" do
-      obj = R10K::Module.new('foo', '/modulepath', :git => 'git url')
-      expect(obj).to be_a_kind_of(R10K::Module::Git)
+    [
+      ['foo', {git: 'git url'}],
+      ['foo', {type: 'git', source: 'git url'}],
+    ].each do |scenario|
+      it "accepts a name matching #{scenario[0]} and args #{scenario[1].inspect}" do
+        obj = R10K::Module.new(scenario[0], '/modulepath', scenario[1])
+        expect(obj).to be_a_kind_of(R10K::Module::Git)
+        expect(obj.send(:instance_variable_get, :'@remote')).to eq('git url')
+      end
     end
   end
 
   describe 'delegating to R10K::Module::Forge' do
     [
       ['bar/quux', nil],
       ['bar-quux', nil],
+    ].each do |scenario|
+      it "accepts a name matching #{scenario[0]} and args #{scenario[1].inspect}" do
+        obj = R10K::Module.new(scenario[0], '/modulepath', scenario[1])
+        expect(obj).to be_a_kind_of(R10K::Module::Forge)
+      end
+    end
+    [
       ['bar/quux', '8.0.0'],
+      ['bar/quux', {type: 'forge', version: '8.0.0'}],
     ].each do |scenario|
       it "accepts a name matching #{scenario[0]} and args #{scenario[1].inspect}" do
-        expect(R10K::Module.new(scenario[0], '/modulepath', scenario[1])).to be_a_kind_of(R10K::Module::Forge)
+        obj = R10K::Module.new(scenario[0], '/modulepath', scenario[1])
+        expect(obj).to be_a_kind_of(R10K::Module::Forge)
+        expect(obj.send(:instance_variable_get, :'@expected_version')).to eq('8.0.0')
       end
     end
   end
```

Comment:
Should there be svn tests? I notice there aren't any in this file and you didn't add any in `spec/unit/module/svn_spec.rb`.

Reply:
May as well. Adding.